pkg/asset/cluster/cluster.go

```diff
@@ -49,8 +50,14 @@ func (c *Cluster) Generate(parents map[asset.Asset]*asset.State) (*asset.State,
 		return nil, fmt.Errorf("failed to get terraform.tfvar state in the parent asset states")
 	}
 
-	// Copy the terraform.tfvars to the working dir for the terraform.
-	if err := ioutil.WriteFile(filepath.Join(dir, state.Contents[0].Name), state.Contents[0].Data, 0600); err != nil {
+	// Copy the terraform.tfvars to a temp directory where the terraform will be invoked within.
+	tmpDir, err := ioutil.TempDir(os.TempDir(), "openshift-install-")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create temp dir: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
```

Comment:
Do we really want to unconditionally remove the terraform state? We could gate this with an option to keep tempfiles if someone wishes to, so we can use it for destruction later

Reply:
@steveeJ We still have the tfstate file in the assets dir, it will be persisted by the store
https://github.com/openshift/installer/blob/master/cmd/openshift-install/main.go#L47
https://github.com/openshift/installer/blob/master/cmd/openshift-install/main.go#L66